File: ELEC881_Copilot/src/Function Name Masked/Python/q26-split-array-largest-sum.py
class sol(object):
    def splitArray(self, nums, k):
        def valid(mid):
            count = 1
            total = 0
            for num in nums:
                total += num
                if total > mid:
                    total = num
                    count += 1
            return count <= k

        left, right = max(nums), sum(nums)
        while left < right:
            mid = (left + right) // 2
            if valid(mid):
                right = mid
            else:
                left = mid + 1
        return left

File: ELEC881_Copilot/src/Function Name Masked/Java script (JS)/q26-split-array-largest-sum.js
var abc = function(nums, k) {
    let left = Math.max(...nums);
    let right = nums.reduce((a, b) => a + b);
    while (left < right) {
        let mid = left + Math.floor((right - left) / 2);
        let sum = 0;
        let count = 1;
        for (let num of nums) {
            sum += num;
            if (sum > mid) {
                sum = num;
                count++;
            }
        }
        if (count > k) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }
    return left;
}